### DISCUSSION TITLE
Using notFound() Doesn't Return a 404?

### DISCUSSION THREAD

**Discussion by darrylmorley** [vercel/next.js#55431] [Help] [answered]
Hi,

I'm using notFound() with a custom not-found page. I'm not seeing a 404 for the title or network response.

If I remove my custom not-found page, the page title becomes `404:This Page Could Not Be Found` but the network response is 200.

Is this normal behaviour and how should I return a 404?

**Comment by icyJoseph** [comment]
Could you share the folder structure?

**Reply by darrylmorley** [reply]
Sure, the below is in the app directory

![Screenshot 2023-09-15 115115](https://github.com/vercel/next.js/assets/20406137/125ef9c7-505a-4a19-9e29-42febcafd1ee)

**Reply by icyJoseph** [reply]
Aha, maybe you are seeing this:

> Along with serving a custom UI, Next.js will return a 200 HTTP status code for streamed responses, and 404 for non-streamed responses.

**Reply by darrylmorley** [reply]
So removing loading,jsx & removing any trace of Suspense does not result in a 404 being returned from notFound(). I have no idea how you are supposed to be able to return a 404 using app directory. 🤷‍♂️

**Reply by icyJoseph** [reply]
Mmm, how are you doing this? Because for example I use it here:

- https://github.com/icyJoseph/icyJoseph/blob/dev/src/app/blog/%5Bslug%5D/page.tsx#L67

And that gives me back a 404 when I got an invalid page.

**Reply by darrylmorley** [reply, marked as answer]
I managed to get it working yesterday. I don't know why removing loading.jsx from my project initially had no effect. But after several attempts with different configurations, all is now fine. It was indeed completely related to using loading.jsx. 

Thanks for your help.

**Reply by icyJoseph** [reply]
No worries! Don't forget to add an answer mark too. I think your last comment is a good answer for this discussion.
